lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -129,11 +129,18 @@ def merged_group_access(self):
                             # special case - means somewhere up the hierarchy, merged access rules have eliminated
                             # all group_ids from this partition, so there's no possible intersection.
                             continue
-                        # otherwise, if the parent defines group access rules for this partition,
-                        # intersect with the local ones.
-                        merged_access[partition_id] = list(
-                            set(merged_access[partition_id]).intersection(group_ids)
-                        ) or False
+                        # For Feature Based Enrollments, we want to allow granting access
+                        # to all graded problems in a unit at the unit level
+                        if partition_id == 51 and parent.scope_ids.block_type == 'vertical':
```

Comment:
This is a) super brittle to changes in the course structure (`sequence` > `vertical` is just the structure enforced by studio, but you can import xml that's different), and b) the partition id shouldn't be hardcoded.